conf/broker.conf

```diff
@@ -102,6 +102,12 @@ tlsAllowInsecureConnection=false
 # Using a value of 0, is disabling unackeMessage limit check and consumer can receive messages without any restriction 
 maxUnackedMessagesPerConsumer=50000
 
+# Max number of unacknowledged messages allowed per shared subscription. Broker will stop dispatching messages to
+# all consumers once, this limit reaches until consumer starts acknowledging messages back and unack count reaches
+# to maxUnackedMessagesPerDispatcher/2. Using a value of 0, is disabling unackedMessage-limit check and dispatcher
+# can dispatch messages without any restriction
+maxUnackedMessagesPerDispatcher=200000
```

Comment:
That's a different thing. Here we're talking about `pendingAcks` for shared subscriptions. The dispatcher needs a way to know which messages were sent to a particular consumer, to be able, in case the consumer disconnect, to redeliver the messages somewhere else. 

For short term we could just use this per-subscription limit and switch to `ConcurrentLongLongPairHashMap` that it's already implemented `(long,long)  --> (long,long)`. We don't need the 2nd long in the value, but the overhead is not that big. 

That way, even keeping 10 M of pending acks would require only 320 MB of memory and would not cause GC troubles.